frontend/src/component/common/ConstraintsList/StrategyEvaluationItem/StrategyEvaluationItem.tsx

```diff
@@ -35,45 +39,33 @@ const StyledType = styled('span')(({ theme }) => ({
     width: theme.spacing(10),
 }));
 
-const StyledValuesGroup = styled('ul')(({ theme }) => ({
-    display: 'flex',
-    flexFlow: 'row wrap',
-    alignItems: 'center',
-    gap: theme.spacing(0.5),
-    listStyle: 'none',
-    padding: 0,
-}));
-
-const StyledValue = styled('li')(({ theme }) => ({
-    [`.${disabledStrategyClassName} &`]: {
-        filter: 'grayscale(1)',
-        color: theme.palette.text.secondary,
-    },
-    ':not(&:last-of-type)::after': {
-        content: '", "',
-    },
-}));
-
 /**
  * Abstract building block for a list of constraints, segments and other items inside a strategy
  */
-export const StrategyEvaluationItem: FC<StrategyItemProps> = ({
+export const StrategyEvaluationItem: FC<StrategyEvaluationItemProps> = ({
     type,
     children,
     values,
+    onSetTruncated,
 }) => (
     <StyledContainer>
         <StyledType>{type}</StyledType>
         <StyledContent>
             {children}
-            {values && values?.length > 0 ? (
-                <StyledValuesGroup>
-                    {values?.map((value, index) => (
-                        <StyledValue key={`${value}#${index}`}>
-                            {value}
-                        </StyledValue>
-                    ))}
-                </StyledValuesGroup>
+            {values && values?.length === 1 ? (
+                <Truncator
+                    title={values[0]}
+                    arrow
+                    lines={2}
+                    onSetTruncated={() => onSetTruncated?.(false)}
+                >
+                    {values[0]}
+                </Truncator>
+            ) : null}
+            {values && values?.length > 1 ? (
+                <Truncator title='' lines={2} onSetTruncated={onSetTruncated}>
+                    {values.join(', ')}
+                </Truncator>
             ) : null}
         </StyledContent>
```

Comment:
I didn't find any reliable way of truncating many long values when each is wrapped in it's own chip. It's far easier to do it for strings.

Reply:
Ah, mm. Right. I see what you mean. At the same time, this makes the structure less clear to screen readers etc. It'd be nice to retain it as a list if possible, but I'm not gonna halt the PR for it. When it's not a list, there's no indication in the rendered html that it's not just one long value with commas in it. If it's a list, you'll typically have screen readers announce how many items it contains and then go through them in order (as far as I understand).

So what is the actual issue here? What are you truncating? "all the items"? I'd think the list itself has an intrinsic width that you could check? Probably?  I'd think the truncator handles that, but maybe it don't 🤷🏼

Reply:
Valid point. I'll experiment I there is an approach that will be the best of both words.